app/controller/command/commands/team.py

```diff
@@ -385,19 +387,29 @@ def add_helper(self, param_list, user_id) -> ResponseTuple:
         """
         try:
             command_user = self.facade.retrieve(User, user_id)
-            teams = self.facade.query(Team, [('github_team_name',
-                                              param_list['team_name'])])
-            if len(teams) != 1:
-                return self.lookup_error, 200
-            team = teams[0]
+            command_team = param_list['team_name']
+            team = get_team_by_name(self.facade, command_team)
             if not check_permissions(command_user, team):
                 return self.permission_error, 200
 
             user = self.facade.retrieve(User, param_list['username'])
             team.add_member(user.github_id)
             self.gh.add_team_member(user.github_username, team.github_team_id)
             self.facade.store(team)
-            msg = "Added User to " + param_list['team_name']
+
+            # If this team is a team with special permissions, promote the user
+            promoted_level = Permissions.member
+            if command_team == self.config.github_team_admin:
+                promoted_level = Permissions.admin
+            elif command_team == self.config.github_team_leads:
+                promoted_level = Permissions.team_lead
+
+            msg = "Added User to " + command_team
+            if promoted_level != Permissions.member:
+                logging.info(f"Promoting {command_user} to {promoted_level}")
+                user.permissions_level = promoted_level
+                self.facade.store(user)
+                msg += f" and promoted user to {promoted_level}"
```

Comment:
Still runs into the issue.

Say someone needs to be admin and is also a team lead for some team. An existing admin adds them into the admin team because they are admin. That person gets promoted to admin privileges. Now the existing admin adds them into the team leads team because they are also team lead. That person gets demoted to lead privileges.

Reply:
838444538a1b7d3160929b901e04851eea3326c9 and 5e7013d25f7e862ea910399badf160efe3bb9751